src/compiler/objective_c_generator.cc

```diff
@@ -176,65 +176,71 @@ void PrintMethodImplementations(Printer *printer,
 
 string GetHeader(const ServiceDescriptor *service, const string prefix) {
   string output;
-  grpc::protobuf::io::StringOutputStream output_stream(&output);
-  Printer printer(&output_stream, '$');
+  {
+    // Scope the output stream so it closes and finalizes output to the string.
```

Comment:
> That is supposed to be a warning.

I don't think that's true; taking the address of an automatic variable is a perfectly acceptable thing to do. A very common use case of it is to pass the address of a local variable to a function call (e.g. `&error`), as "output parameters." I'd think passing it to a constructor is as fine.

> which is against the spec

Are we sure of that? The conversation in StackOverflow seems to me to imply the spec authors decided to allow it. Without that, there's no way to move on return, or you'd destruct your object before moving it.

In any case, back to this PR: let's make clear in the comment that without the scope, the string might get moved before the printer flushes _which results in a crash_ (even if it ends up being some compiler's bug).